frontend/src/App.js

```diff
@@ -14,6 +15,28 @@ import './App.less';
 
 const INACTIVE_TIMEOUT = 3600000;
 
+function persist(key, value) {
+  window.localStorage.setItem(key, JSON.stringify(value));
+}
+
+const dateFormat = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/;
+
+function jsonReviver(key, value) {
+  if (typeof value === 'string' && dateFormat.test(value)) {
```

Comment:
Since we are only storing history elements I think we could simplify the regexp to 
`if (key === 'datetime')`

Reply:
In my opinion, it's better to have a common solution for datatime fields than relying on the particular structure of currently saved data.